File: CoreDataExample/Card.m
//
//  Card+CoreDataClass.m
//  CoreDataExample
//
//  Created by Andrew Riznyk on 7/12/17.
//  Copyright © 2017 Andrew Riznyk. All rights reserved.
//

#import "Card.h"
#import "Deck.h"

@implementation Card

- (NSString*) entityName {
    return @"Card";
}

+ (NSString*) entityName {
    return @"Card";
}

+ (id) insertIntoContext:(NSManagedObjectContext*)context {
    return [self insertNewObjectIntoContextHelper:context];
}

+ (id) insertNewObjectIntoContextHelper:(NSManagedObjectContext*)context {
    return [NSEntityDescription insertNewObjectForEntityForName:[self entityName] inManagedObjectContext: context];
}


@end

File: CoreDataExampleSwift/Card.swift
//
//  Card+CoreDataClass.swift
//  CoreDataExampleSwift
//
//  Created by Andrew Riznyk on 7/30/17.
//  Copyright © 2017 Andrew Riznyk. All rights reserved.
//

import Foundation
import CoreData

protocol ManagedObjectType {
    static var entityName: String { get }
}

@objc(Card)
public class Card: NSManagedObject {

    @NSManaged public var cardID: Int64
    @NSManaged public var deck_id: Int64
    @NSManaged public var number: Int32
    @NSManaged public var type: Int32
    @NSManaged public var deck: Deck?
    @NSManaged public var random: String?
}

extension Card: ManagedObjectType {
    static let entityName = "Card"
}
